mcs/class/corlib/System.IO/LogcatTextWriter.cs

```diff
@@ -46,34 +48,22 @@ public override void WriteLine ()
 			var o = line.ToString ();
 			line.Clear ();
 
-			Log (LogLevel.Info, appname, o);
+			unsafe {
+				fixed (byte *b_appname = appname)
+				fixed (byte *b_message = Encoding.UTF8.GetBytes(o)) {
+					Log (b_appname, 1 << 5 /* G_LOG_LEVEL_MESSAGE */, b_message);
```

Comment:
This line is troubling to me: `b_message` will *not* be `\0`-terminated, and we call `Log()` with it, which is [`ves_icall_System_IO_LogcatTextWriter_Log ()`](https://github.com/mono/mono/pull/6083/files#diff-eba0fd0052bf3ec9dd60f2f3e90f19e1R7902), which calls `g_log()` with `b_message` as `message`, and -- again -- `message` won't be `\0`-terminated.

This is *probably* why [things go kablooey](http://xqa.blob.core.windows.net/gist/report-4e0d0e63489d4f60b06c5077434d84a6.txt) with the xamarin-android + mono/2017-12 integration bump:

```
backtrace:
    #00  pc 0002702e  /system/lib/libc.so (__vfprintf+269)
    #01  pc 000268bb  /system/lib/libc.so (vasprintf+66)
    #02  pc 002b9cf0  /data/app-lib/Xamarin.Android.JcwGen_Tests-1/libmonosgen-2.0.so (monoeg_g_log+80)
```

Reply:
👍

Reply:
https://github.com/mono/mono/pull/6954